layer/migration.go

```diff
@@ -178,6 +173,18 @@ func (ls *layerStore) RegisterByGraphID(graphID string, parent ChainID, tarDataF
 		referenceCount: 1,
 		layerStore:     ls,
 		references:     map[Layer]struct{}{},
+		diffID:         diffID,
+		size:           size,
+		chainID:        createChainIDFromParent(parent, diffID),
+	}
+
+	ls.layerL.Lock()
```

Comment:
If when this is run there will not be contention for the adding layers then I am fine with leaving it like this.  Two locks mean two existence checks and that is not ideal.